transport/http/src/main/java/org/kaazing/gateway/transport/http/bridge/filter/HttpLoginSecurityFilter.java

```diff
@@ -288,24 +288,34 @@ protected boolean login(NextFilter nextFilter,
                      rolesAreSufficient);
         }
 
+        String clientChallengeScheme = authToken.getScheme();
+        String expectedChallengeScheme = getBaseAuthScheme(realm.getChallengeScheme());
+
         // We have to issue a challenge when ALL of the below hold:
         //
         // - no authorization token was presented
         // - not all required roles (if any) are present
         //
         // Issue a challenge since authentication is required.
-        if (authTokenIsMissing(authToken) &&
-            rolesAreSufficient == false) {
+        if ((authTokenIsMissing(authToken) &&
+            rolesAreSufficient == false) ) {
             return loginMissingToken(nextFilter, session, httpRequest, authToken, additionalCallbacks, realms, realmIndex, loginContexts);
         }
 
         DefaultLoginResult loginResult = null;
+        ResultAwareLoginContext loginContext = null;
+
+        if (clientChallengeScheme != null && clientChallengeScheme.equalsIgnoreCase(expectedChallengeScheme) == false) {
+            String challenge = sendChallengeResponse(nextFilter, session, httpRequest, new DefaultLoginResult(), realms, realmIndex, loginContexts);
+            if (loggerEnabled()) {
+                log(String.format("Login module login failed; Issued another challenge '%s'", challenge));
```

Comment:
Is this the best message?  Seems to me like it is sending the challenge because the scheme no longer matches.